README.md

```diff
@@ -17,22 +17,44 @@ All mutability can be _piped out_ of your code.
 use bevy::prelude::*;
 use bevy_pipe_affect::prelude::*;
 
-fn main() {
+fn main() -> AppExit {
     App::new()
         .add_plugins(DefaultPlugins)
         .add_systems(
             Update,
             pure(rainbow_clear_color) // pure() is optional, just forces the system to be read-only
                 .pipe(affect),
         )
-        .run();
+        .run()
 }
 
 /// This system defines the clear color as a pure function of time.
-fn rainbow_clear_color(time: Res<Time>) -> impl Effect + use<> {
+fn rainbow_clear_color(time: Res<Time>) -> ResSet<ClearColor> {
     let color = Color::hsv(time.elapsed_secs() * 20.0, 0.7, 0.7);
     res_set(ClearColor(color))
 }
+
+// One benefit of writing systems as pure function is that testing becomes much easier
+#[cfg(test)]
+mod tests {
+    use bevy::ecs::system::RunSystemOnce;
+
+    use super::*;
+
+    #[test]
+    fn test_rainbow_clear() {
+        // Create a world to test against
+        let mut world = World::new();
+        world.insert_resource::<Time>(Time::default());
+
+        // Now that we have a Res<Time>, we can run our system and get an output
+        let output = world.run_system_once(rainbow_clear_color).unwrap();
+
+        // Then, instead of checking that the world and it's resources changed in a particular way,
+        // all we have to do is check that output has the proper value
```

Comment:
```suggestion
        // Then, instead of checking that the world and its resources changed in a particular way,
        // all we have to do is check that the output is correct
```

minor grammar things